browser/brave_shields/ad_block_service_browsertest.cc

```diff
@@ -651,15 +651,98 @@ IN_PROC_BROWSER_TEST_F(AdBlockServiceTest, CnameCloakedRequestsGetBlocked) {
   ASSERT_EQ(4ULL, inner_resolver->num_resolve());
 
   // XHR request directly to a blocked third-party endpoint.
-  // The resolver should still be queried once for this request, although it
-  // doesn't need to be.
-  // See https://github.com/brave/brave-browser/issues/15302
+  // The resolver should not be queried for this request.
```

Comment:
can we also test something that would be resolved in normal profile but won't be resolved in tor profile?
you can use
`brave::NewOffTheRecordWindowTor` + `BrowserList::GetInstance()->GetLastActive()->profile()`
or 
`TorProfileManager::SwitchToTorProfile`

Reply:
@darkdh that is a great idea!

I'm using this to get the Tor `WebContents` in a test that's otherwise functionally identical to `AdblockServiceTest.CnameCloakedRequestsGetBlocked`:

```cpp
brave::NewOffTheRecordWindowTor(browser());
content::RunAllTasksUntilIdle();

Browser* tor_browser = BrowserList::GetInstance()->GetLastActive();
EXPECT_TRUE(tor_browser->profile()->IsTor());

ui_test_utils::NavigateToURL(tor_browser, tab_url);

content::WebContents* contents =
    tor_browser->tab_strip_model()->GetActiveWebContents();
```

Unfortunately the `NavigateToURL` call is timing out, and `EvalJs` is throwing some CSP errors (which might just be because it's running on the Tor NTP).

Do you know how to get the tab navigation working? Otherwise it may be better to defer this to a followup.

Reply:
Not at the moment so let's do it as follow up